### DISCUSSION TITLE
Show custom message if user does not have access to the panel instead of 403 FORBIDDEN

### DISCUSSION THREAD

**Discussion by shojibur** [filamentphp/filament#9224] [Help] [answered]
Hey guys

Anyone could help me to find out how i can show custom message or even redirect to other page if currently logged in user does not have access to current panel instead of following message ? 

<img width="1344" alt="Screenshot 2023-10-21 at 12 21 54 AM" src="https://github.com/filamentphp/filament/assets/8462446/dccdde0b-3753-4248-aedc-1dea8f3e8838">

**Comment by VincvanAs** [comment, marked as answer]
php artisan vendor:publish --tag=laravel-errors and customize the error pages in view/errors

**Reply by adrianlzx1996** [reply]
Hey @VincvanAs ,  thank you for your help.

I have tried your suggestion.

However i have another problem. 
By using the following code, I would expect the page to show the "You do not own this post.".

The 403 page is still showing 403, Forbidden.

```php
use App\Models\Post;
use App\Models\User;
use Illuminate\Auth\Access\Response;
 
/**
 * Determine if the given post can be updated by the user.
 */
public function update(User $user, Post $post): Response
{
    return $user->id === $post->user_id
                ? Response::allow()
                : Response::deny('You do not own this post.');
}
```

https://laravel.com/docs/10.x/authorization#policy-responses

**Reply by VincvanAs** [reply]
Where did you place this code, and what action is it returning the default response

**Reply by adrianlzx1996** [reply]
This code was placed in Resource Policy, for example App\Policies\UserPolicy.php.

It is returning the default response, when my user do not have the necessary permission for that action.

**Comment by leandrocfe** [comment]
or 
`php artisan make:view errors/403`

```html
<!-- resources/views/errors/403.blade.php -->
<div>
    custom message...
</div>
```

**Comment by PetroZaburko** [comment]
You can override method `authorizeResourceAccess()` in your resource pages (Create, Edit... or Manage page if you use simple page) like this:
```php
public static function authorizeResourceAccess(): void
{
     abort_unless(static::getResource()::canAccess(), 404);
}
```

and change response code (in my case I set 404). Now you can create custom error view according your error code.
And no one knows about the existence of your closed resource
